zenodo/modules/pages/views.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Zenodo pages."""
+
+from __future__ import absolute_import, print_function
+
+from flask import Blueprint, current_app, flash, redirect, render_template, \
+    request
+from flask_babelex import lazy_gettext as _
+from flask_mail import Message
+from flask_security import current_user
+
+from .forms import ContactForm
+from .utils import send_support_email
+
+blueprint = Blueprint(
+    'zenodo_pages',
+    __name__,
+    template_folder='templates',
+    static_folder='static',
+)
+
+
+@blueprint.route(
+    '/supportform',
+    methods=['GET', 'POST']
+)
+def support_form():
+    """Render contact form."""
+    form = ContactForm()
+
+    """If form is validated send email to the admin."""
+    if form.validate_on_submit():
```

Comment:
Hey, form is not sending anything on POST now when logged-in, because ``form.email`` is not set before the validation is checked.
If you insert ipdb before this line and check ``form.errors`` you'll get:

``{'email': ['This field is required.']}``